src/TreeViewList/TreeViewList.tsx

```diff
@@ -152,6 +179,21 @@ const TreeViewList = <T,>({
       </TooltipTreeItem>
     ));
 
+  // expand nodes that match the search term
+  useEffect(() => {
+    if (expandSearchTerm) {
+      // Reset expanded ids
+      setExpanded([]);
+
+      // if the search term is not empty, get the ids of the matching nodes and set them as expanded
+      if (searchTerm !== "") {
+        const ids = searchTree(parameters, searchTerm);
+        setExpanded(ids);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should not add this to avoid a dependency warning instead you can wrap `searchTree` in a useCallBack hook to fix it properly.